### PR TITLE
feat: refactor sidecarTopSlot for any component

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

This PR lightly refactors the `SidebarSidecarLayout` in order to accommodate content of arbitrary height in the `sidecarTopSlot`.

## 🤷 Why

Upstream, the `sidecarTopSlot` requires its content to have a height of exactly `TODO get this number`. This was implemented as a fix for the variant selector positioning, and is working just fine since at present we only use `sidecarTopSlot` for our variant selector.

Looking ahead, we want to ensure our layout can handle any content in that slot area. Updating the `sidecarTopSlot` to support content with an arbitrary height is feasible, will give us better layout flexibility for future work, and will reduce magic numbers in our code, so it feels worth implementing.

## 🛠️ How

> **Note**: a good portion of the lines-of-code changed in this PR are for setting up 🚧 [/test-sidecar-layout], which will be removed before merging.

- Adds [/test-sidecar-layout] to demo layout issue with arbitrary `sidecarTopSlot` content height
- Refactors `sidecarTopSlot` placement to avoid the issue
- ✅ Remove the [/test-sidecar-layout] page
    - Will do this once this PR has been reviewed, approved, and is ready to merge!

## 📸 Design Screenshots

- Before: [before/test-sidecar-layout]
- After: [/test-sidecar-layout]

| Before | After |
| - | - |
| before | after |

## 🧪 Testing

- [ ] Confirm the layout issue exists upstream
    - Here's [a preview link of `/test-sidecar-layout` before the layout change][before/test-sidecar-layout]
    - The layout works when the height is exactly `63px`, but content is cut off when height is any larger
- [ ] Inspect the revised test page at [/test-sidecar-layout], and confirm the `sidecarTopSlot` can now accommodate content of arbitrary height
- [ ] Visit some non-test pages to confirm they still behave as expected
    - Long sidecar example: [/nomad/docs/upgrade/upgrade-specific]

## 💭 Anything else?

Not at the moment!

[task]: https://app.asana.com/0/1204333057896641/1204669563792095/f
[preview]: https://dev-portal-git-zsrefactor-sidecar-for-noscroll-slot-hashicorp.vercel.app/
[/test-sidecar-layout]: https://dev-portal-git-zsrefactor-sidecar-for-noscroll-slot-hashicorp.vercel.app/test-sidecar-layout
[before/test-sidecar-layout]: https://dev-portal-btf10hd93-hashicorp.vercel.app/test-sidecar-layout
[/nomad/docs/upgrade/upgrade-specific]: https://dev-portal-git-zsrefactor-sidecar-for-noscroll-slot-hashicorp.vercel.app/nomad/docs/upgrade/upgrade-specific

### CODE REVIEW

File: src/components/dev-height-adjustable-placeholder/index.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { useState } from 'react'
+
+function DevHeightAdjustablePlaceholder({
```
Comment (zchsh) [inline src/components/dev-height-adjustable-placeholder/index.tsx:3]: Found this useful in testing, but will delete along with `/test-sidecar-layout` page before merging 👍

File: src/layouts/sidebar-sidecar/components/sidecar-scroll-container/sidecar-scroll-container.module.css [hunk 2]
```diff
@@ -14,9 +14,7 @@
 	display: flex;
 	flex-direction: column;
 	isolation: isolate;
-
-	/* Note: max-sidecar-height is set in sidebar-sidecar-layout.module.css. */
-	max-height: var(--max-sidecar-height);
+	min-height: 0;
```
Comment (zchsh) [inline src/layouts/sidebar-sidecar/components/sidecar-scroll-container/sidecar-scroll-container.module.css:17]: `sidecar-scroll-container` is now within a flex-column layout, and the enclosing layout handles the max-height concern. We need `min-height: 0` here to ensure the `sidecar-scroll-container` starts at zero and grows to fill the flex column.

File: src/layouts/sidebar-sidecar/index.tsx [hunk 3]
```diff
@@ -114,13 +114,8 @@ const SidebarSidecarLayoutContent = ({
 								/>
 							) : null}
 						</main>
-						<div
-							className={classNames(
-								s.sidecarPosition,
-								sidecarTopSlot && s.topSlotDropdown
```
Comment (zchsh) [inline src/layouts/sidebar-sidecar/index.tsx:120]: `s.topSlotDropdown` class is no longer needed, so removed.

File: src/pages/test-sidecar-layout/index.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+				fullPath: '/packer/downloads',
+			},
+		],
+		showFilterInput: false,
+		title: 'Packer',
+		visuallyHideTitle: true,
+	},
+] as $TSFixMe
+
+export default function TestSidecarLayout() {
```
Comment (zchsh) [inline src/pages/test-sidecar-layout/index.tsx:112]: Will delete this test page before merging!

File: src/layouts/sidebar-sidecar/sidebar-sidecar-layout.module.css [hunk 5]
```diff
@@ -151,40 +152,37 @@ for further details.
...
+.sidecarTopSlot {
+	flex-shrink: 0;
+
+	/* Note: we don't ever want the top slot to be very tall, and we don't
+	   really want this container to ever have to scroll.
+	   But we do want to gracefully handle unexpected edge cases where it 
+		 does exceed the available height. We set its max-height to
+		 half the available sidecar max-height, as otherwise it would clobber
+		 the sidecar scrollable content down to a 0px tall container. */
+	max-height: calc(var(--max-sidecar-height) / 2);
```
Comment (zchsh) [inline src/layouts/sidebar-sidecar/sidebar-sidecar-layout.module.css:182]: Added a comment here to clarify why this is needed - TLDR for expected use cases we want `sidecarTopSlot` to never have overflow, but in the unexpected edge case that `sidecarTopSlot` has tall content, we fall back to splitting the available space with the scrolling part of the sidecar 50/50.

**Comment by zchsh** [comment]
@BrandonRomano tagging ya here since I feel like you might already have a good deal of context 🙇‍♂️

**Comment by BrandonRomano** [comment]
Thanks @zchsh I'll be sure to check this out shortly

**Review by BrandonRomano — APPROVED** [review]
Wow, your test page here is brilliant, and the solution itself is easy to follow.

This is great 👏 👏 👏

**Pull request merged**